modules/core/shared/src/main/scala/eu/timepit/refined/numeric.scala

```diff
@@ -181,3 +181,84 @@ private[refined] trait NumericInference {
   ): Greater[A] ==> Greater[B] =
     Inference(nc.gt(wa.value, nc.fromInt(tb())), s"greaterInferenceWitNat(${wa.value}, ${tb()})")
 }
+
+private[refined] trait NumericMin {
```

Comment:
@howyp What do you think about moving all `Min` / `Max` instances into the `Min` / `Max` companions? That would make them self-contained and would mean that you don't need to import `numeric._` to get instances.

Reply:
Generally I much prefer that approach than knowing you have to import certain wildcards. But given you have to import things like `numeric._` for most features in refined, would it be consistent? Because the `Min`/`Max` instance are captured at the companion object creation site, users of `MaxValue` wouldn't need any imports from refined.

It makes me wonder whether it would be a good idea for #386 to look at providing imports which give you everything in refined. At the moment, I think defining a refined type requires two package imports and at least one type import? Or did I miss something?

Reply:
The über-import issue is tracked in #184 and I agree it is a good idea.

The current organization of type class instances grew historically and if I would start from fresh I probably would add the `Validate` and `Inference` instances to the companions of the respective predicates. The same could be done for `Min` and `Max` instances in this setting, although I could also imagine putting them in the `Min` / `Max` companions since these type classes are orthogonal to refined's main functionality and its other type classes. `Min` / `Max` also make only sense for numeric types and therefore are narrower in scope than e.g. `Validate`.

I haven't reached a conclusion on this matter yet. Maybe the consistency argument is good enough to leave it as is.

Reply:
OK, I think that's enough to convince me to move them into the companion objects. Especially that there is some overlap between the `numeric` and `char` packages in that `Min`/`Max` support both (as well as particular `boolean` predicates).

I'll move them over unless you say otherwise.